bcl/src/bc_esp8266.c

```diff
@@ -863,3 +973,40 @@ static void _bc_esp8266_set_rtc_time(char *str)
     }
     bc_rtc_set_date_time(&rtc);
 }
+
+bool bc_esp8266_check_ap_availability(bc_esp8266_t *self)
+{
+    if (self->_state != BC_ESP8266_STATE_DISCONNECTED || self->_config.ssid[0] == '\0')
+    {
+        return false;
+    }
+
+    _bc_esp8266_enable();
+
+    self->_state = BC_ESP8266_STATE_INITIALIZE;
+    self->_state_after_init = BC_ESP8266_STATE_AP_AVAILABILITY_OPT_COMMAND;
+
+    bc_scheduler_plan_now(self->_task_id);
+
+    return true;
+}
+
+bool bc_esp8266_get_ap_availability_result(bc_esp8266_t *self, bool *available, int *rssi)
+{
+    if (self->_state != BC_ESP8266_STATE_AP_AVAILABILITY_RESPONSE)
+    {
+        return false;
+    }
+
+    if (self->_message_buffer[0] == '1')
```

Comment:
I have a better idea now what is `1` doing, but not sure if this is the clean approach. In the `BC_ESP8266_STATE_AP_AVAILABILITY_RESPONSE` state you set `self->_message_buffer` with preceding `1` character if the network and RSSI value is available but you don't set it otherwise. Most likely I am overlooking something, but the solution is not very obvious contrary to the rest of the implementation :) Why don't we simply parse RSSI already in the state handler and store it in `self->_rssi` with a boolean `self->_network_found` flag?